otcextensions/sdk/dws/v1/snapshot.py

```diff
@@ -0,0 +1,69 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from openstack import resource
+from otcextensions.sdk.dws.v1 import cluster as _cluster
+
+
+class Snapshot(resource.Resource):
+    base_path = '/snapshots'
+
+    resource_key = 'snapshot'
+    resources_key = 'snapshots'
+
+    # capabilities
+    allow_create = True
+    allow_fetch = True
+    allow_delete = True
+    allow_list = True
+
+    # Properties
+    #: ID of the cluster for which snapshots are created.
+    cluster_id = resource.Body('cluster_id')
+    #: Snapshot name.
+    name = resource.Body('name')
+    #: Snapshot description.
+    description = resource.Body('description')
+    #: Time when a snapshot starts to be created.
+    started = resource.Body('started')
+    #: Time when a snapshot is complete.
+    finished = resource.Body('finished')
+    #: Snapshot size, in GB.
+    size = resource.Body('size', type=float)
```

Comment:
even docs explicitly says that, all examples are just integers. Can you please verify?

Reply:
Its specified as integer but resp body returns float size  :(
will also raise this as Bug

RESP BODY: {"snapshot":{"id":"XXXXXXXXX","name":"XXXXXXX","description":"","started":"2023-01-30T10:14:36","finished":"2023-01-30T10:19:03",**"size":0.403551,**"status":"AVAILABLE","type":"AUTO","cluster_id":"XXXX"}}